lib/src/command/outdated.dart

```diff
@@ -67,101 +72,123 @@ class OutdatedCommand extends PubCommand {
-            ..sort(argResults['pre-releases'] ? null : Version.prioritize);
-      final upgradable = upgradablePackages
-          .firstWhere((id) => id.name == name, orElse: () => null)
-          ?.version;
-      final resolvable = resolvablePackages
-          .firstWhere((id) => id.name == name, orElse: () => null)
-          ?.version;
-      final latest = available.last;
-      final description = packageRef.description;
+      final current = (entrypoint.lockFile?.packages ?? {})[name];
+
+      final upgradable = upgradablePackages.firstWhere((id) => id.name == name,
+          orElse: () => null);
+      final resolvable = resolvablePackages.firstWhere((id) => id.name == name,
+          orElse: () => null);
+
+      // Find the latest version, and if it's overridden.
+      var latestIsOverridden = false;
+      PackageId latest;
+      // If not overridden in current resolution we can use this
+      if (!entrypoint.root.pubspec.dependencyOverrides.containsKey(name)) {
+        latest ??= await _getLatest(current);
+      }
+      // If present as a dependency or dev_dependency we use this
+      latest ??= await _getLatest(rootPubspec.dependencies[name]);
+      latest ??= await _getLatest(rootPubspec.devDependencies[name]);
+      // If not overridden and present in either upgradable or resolvable we
+      // use this reference to find the latest
+      if (!upgradablePubspec.dependencyOverrides.containsKey(name)) {
+        latest ??= await _getLatest(upgradable);
+      }
+      if (!resolvablePubspec.dependencyOverrides.containsKey(name)) {
+        latest ??= await _getLatest(resolvable);
+      }
+      // Otherwise, we might simply not have a latest, when a transitive
+      // dependency is overridden the source can depend on which versions we
+      // are picking. This is not a problem on `pub.dev` because it does not
+      // allow 3rd party pub servers, but other servers might. Hence, we choose
+      // to fallback to using the overridden source for latest.
+      if (latest == null) {
+        latest ??= await _getLatest(current ?? upgradable ?? resolvable);
+        latestIsOverridden = true;
+      }
+
       return _PackageDetails(
-          name,
-          await _describeVersion(name, source, description, current),
-          await _describeVersion(name, source, description, upgradable),
-          await _describeVersion(name, source, description, resolvable),
-          await _describeVersion(name, source, description, latest),
-          _kind(name, entrypoint, nonDevDependencies));
+        name,
+        await _describeVersion(
+          current,
+          entrypoint.root.pubspec.dependencyOverrides.containsKey(name),
+        ),
+        await _describeVersion(
+          upgradable,
+          upgradablePubspec.dependencyOverrides.containsKey(name),
+        ),
+        await _describeVersion(
+          resolvable,
+          resolvablePubspec.dependencyOverrides.containsKey(name),
```

Comment:
We should name this bool parameter

Reply:
I wanted to be, but then lines will wrap :(

Reply:
You could assign the override status to a named variable before:
```
final resolvableIsOverridden = resolvablePubspec.dependencyOverrides.containsKey(name);
...
await _describeVersion( resolvable, overridden: resolvableIsOverridden);
```